cmd/icingadb/main.go

```diff
@@ -67,9 +68,17 @@ func run() int {
 
 	heartbeat := icingaredis.NewHeartbeat(ctx, rc, logger)
 	ha := icingadb.NewHA(ctx, db, heartbeat, logger)
+
 	// Closing ha on exit ensures that this instance retracts its heartbeat
 	// from the database so that another instance can take over immediately.
-	defer ha.Close()
+	{
+		defer func() {
+			ctx, cancelCtx := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
Were 3 seconds chosen arbitrarily or is there some reasoning behind it (i.e. gives time for a certain number of retries or something like that)?

Reply:
Arbitrarily.